design-proposals/vm-delete-protection.md

```diff
@@ -0,0 +1,155 @@
+# Overview
+
+VirtualMachine Delete Protection prevents the accidental deletion of VMs from a cluster. As long as deletion protection
+is activated for certain VM objects, they cannot be deleted unless the protection is intentionally removed.
+
+## Motivation
+
+KubeVirt clusters are often managed by automations, CLI commands, 3rd party tools or UI actions, for example to deploy,
+reconfigure or delete VMs. These automations may result in accidental deletion of VMs that should have not been deleted.
```

Comment:
This is not accurate. There is no UI in KubeVirt upstream.

Since both CLI and UI will interact with the extended Kubernetes API to deploy, configure or delete VMs, I'd suggest just to mention that API call to delete the VM(s) will be rejected to avoid accidental deletions of the VMs by sysadmins (as an example of the user scenario).

Reply:
Technically, you can install upstream KubeVirt through HCO on OCP/OKD. :)

Reply:
I will keep it as it is for the reason provided.

Reply:
> Technically, you can install upstream KubeVirt through HCO on OCP/OKD. :)

Is it officially supported?

Let's keep this discussion open because there is no UI in KubeVirt, I just want to get attention of others reading it, altough I understand the importance of mentioning it.

Reply:
Since there seems to be a demand here to delve into the details of the wording, let's do it.
In Wikipedia, the definition of the term [user interface](https://en.wikipedia.org/wiki/User_interface) begins as follows
> In the industrial design field of human–computer interaction, a user interface (UI) is the space where interactions between humans and machines occur. The goal of this interaction is to allow effective operation and control of the machine from the human end, while the machine simultaneously feeds back information that aids the operators' decision-making process.


Contrary to the beginning of the definition of the term [API](https://en.wikipedia.org/wiki/API):
> An application programming interface (API) is a connection between computers or between computer programs. It is a type of software interface, offering a service to other pieces of software.

> There is no UI in KubeVirt upstream.

Yes, the kubevirt/kubevirt repo only provides an API. Never the less there might be humans using kubevirt via the ways which are mentioned in the design proposal, even if the code is in repos other than kubevirt/kubevirt.

From my personal point of view, it is essential that this feature, which is intended to help humans, is seen from the point of view of human users.